src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -233,6 +239,28 @@ void FindBundleHeader()
             return headerOffset != 0;
         }
 
+        private static void CodeSign(string appHostPath)
+        {
+            Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.OSX));
+            const string codesign = @"/usr/bin/codesign";
+            if (!File.Exists(codesign))
+                return;
+
+            var psi = new ProcessStartInfo()
+            {
+                Arguments = $"-s - \"{appHostPath}\"",
+                FileName = codesign,
+                RedirectStandardError = true,
+            };
+
+            using (var p = Process.Start(psi))
+            {
+                p.WaitForExit();
+                if (p.ExitCode != 0)
+                    throw new AppHostSigningException(p.StandardError.ReadToEnd());
```

Comment:
Ideally the exception would also record the exit code (And it would be shown as part of the error message from SDK).